modal/mount.py

```diff
@@ -324,6 +324,9 @@ def add_local_dir(
         condition: Optional[Callable[[str], bool]] = None,
         # add files from subdirectories as well
         recursive: bool = True,
+        # Predicate filter function for file selection, which should accept a filepath and return `True` for inclusion.
+        # Defaults to including all files.
+        ignore: Union[Sequence[str], Callable[[Path], bool]] = [],
     ) -> "_Mount":
```

Comment:
Thinking about it, it might actually make sense to basically copy the Mount add_local* methods into new "private" variants (underscore prefixed) that only have the new ignore arg and not the old condition (and leave the old methods untouched)

Prevents users from starting to use it with Mount that we intend to deprecate, and gives them a reason to switch to the image alternatives.

Reply:
yeah I like it!